### PR TITLE
Alerting: `Add example` dropdown in the Edit notification template view

### PR DESCRIPTION
**Pull request by ppcano**
Fixes https://github.com/grafana/grafana/issues/93686 -> ` Alerting UI: Display common examples of notification templates`  

https://github.com/user-attachments/assets/2b1a7354-27bb-426e-9df6-ab376a390ef0

TODOs

- This PR should be merged after https://github.com/grafana/grafana/pull/94069,  which brings updates to the templating documentation.
- Some examples may not be applicable to data sources-managed alerts. Initially, the dropdown examples is only visible when using the Grafana AlertManager.

### CODE REVIEW

**Comment by brendamuir** [comment]
@ppcano i see the generic "alerts" here - do we wnat to be more specific and use alert instance so we are consistent?

**Comment by ppcano** [comment]
> do we wnat to be more specific and use alert instance so we are consistent?

@brendamuir, this has been a recurring issue, even across the docs. It seems the discussion and decision are still ongoing.

At the moment, I lean toward using the term "alert" for two reasons:
1 - the name of the template variable is `.Alerts` 
2 - the documentation uses the term "alerts" as well:

Screenshot 2024-10-17 at 16 20 09

**Review by soniaAguilarPeiron — APPROVED** [review]
Nice one! 🚀

File: public/app/features/alerting/unified/components/receivers/TemplateForm.tsx [hunk 1]
```diff
@@ -223,20 +232,51 @@ export const TemplateForm = ({ originalTemplate, prefill, alertmanager }: Props)
...
+                                        <Menu.Item
+                                          key={index}
+                                          label={item.description}
+                                          onClick={() => appendExample(item.example)}
+                                        />
+                                      ))}
+                                      <Menu.Divider />
+                                      <Menu.Item
+                                        label={'Examples documentation'}
+                                        url="https://grafana.com/docs/grafana/latest/alerting/configure-notifications/template-notifications/examples/"
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/receivers/TemplateForm.tsx:255]: This one seems to 404 for me; @ppcano can we update this before merging?

**Review by gillesdemey — CHANGES_REQUESTED** [review]

**Review by gillesdemey — APPROVED** [review]

**Pull request merged**